internal/engine/ingester/artifact/artifact.go

```diff
@@ -83,73 +118,229 @@ func (_ *Ingest) Ingest(
 	}, nil
 }
 
-func getApplicableArtifactVersions(
+func (i *Ingest) getApplicableArtifactVersions(
+	ctx context.Context,
 	artifact *pb.Artifact,
 	cfg *ingesterConfig,
 ) ([]map[string]any, error) {
-	var applicableArtifactVersions []struct {
-		Verification   any
-		GithubWorkflow any
-	}
-	// make sure the artifact type matches
+	// Make sure the artifact type matches
 	if newArtifactIngestType(artifact.Type) != cfg.Type {
 		return nil, evalerrors.NewErrEvaluationSkipSilently("artifact type mismatch")
 	}
 
-	// if a name is specified, make sure it matches
+	// If a name is specified, make sure it matches
 	if cfg.Name != "" && cfg.Name != artifact.Name {
 		return nil, evalerrors.NewErrEvaluationSkipSilently("artifact name mismatch")
 	}
 
-	// Build a tag matcher based on the configuration
-	tagMatcher, err := buildTagMatcher(cfg.Tags, cfg.TagRegex)
+	// Get all artifact versions filtering out those that don't apply to this rule
+	versions, err := getAndFilterArtifactVersions(ctx, cfg, i.ghCli, artifact)
 	if err != nil {
 		return nil, err
 	}
 
-	// get all versions of the artifact that are applicable to this rule
-	for _, artifactVersion := range artifact.Versions {
-		if !isProcessable(artifactVersion.Tags) {
-			continue
+	// Get the provenance info for all artifact versions that apply to this rule
+	verificationResults, err := i.getVerificationResult(ctx, cfg, artifact, versions)
+	if err != nil {
+		return nil, err
+	}
+
+	jsonBytes, err := json.Marshal(verificationResults)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make([]map[string]any, 0, len(verificationResults))
+	err = json.Unmarshal(jsonBytes, &result)
```

Comment:
I'm not sure I understand why we're round-tripping `verificationResults` through JSON only to parse it again.

Reply:
You're right, fixed it 👍